### PR TITLE
[VEIOSS-361][JT] Check Your Answer Page

### PR DESCRIPTION
**Pull request by tianjianfeng**

### CODE REVIEW

File: app/controllers/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -18,33 +18,130 @@ package controllers
...
+      val businessSummaryList = getBusinessSummaryList(request, waypoints)
+
+      val salesFromEuSummaryList = getSalesFromEuSummaryList(request, waypoints)
+
+      val containsCorrections = request.userAnswers.get(AllCorrectionPeriodsQuery).isDefined
+
+      val totalVatToCountries =
+        service.getVatOwedToCountries(request.userAnswers).filter(vat => vat.totalVat > 0)
+      val noPaymentDueCountries =
+        service.getVatOwedToCountries(request.userAnswers).filter(vat => vat.totalVat <= 0)
```
Comment (hmrcvmu) [inline app/controllers/CheckYourAnswersController.scala:64]: I think we should be able to use:
```
val (noPaymentDueCountries, totalVatToCountries) = service.getVatOwedToCountries(request.userAnswers).filter(vat => vat.totalVat <= 0)
```

File: app/controllers/CheckYourAnswersController.scala [hunk 2]
```diff
@@ -18,33 +18,130 @@ package controllers
...
+        card = Card(
+          title = Some(CardTitle(content = HtmlContent(messages("checkYourAnswers.correction.heading")))),
+          actions = None
+        )
+      )
+        //.withCssClass("govuk-!-margin-bottom-9").withCard()
+      Seq(
+        (None, businessSummaryList),
+        (None, salesFromEuSummaryList),
+//        (Some("checkYourAnswers.correction.heading"), correctionsSummaryList)
```
Comment (hmrcvmu) [inline app/controllers/CheckYourAnswersController.scala:103]: Should we remove the comments?

File: app/services/SalesAtVatRateService.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  }
+
+  def getTotalVatOwedAfterCorrections(userAnswers: UserAnswers): BigDecimal =
+    getVatOwedToCountries(userAnswers).filter(vat => vat.totalVat > 0).map(_.totalVat).sum
+
+  def getVatOwedToCountries(userAnswers: UserAnswers): List[TotalVatToCountry] = {
+    val vatOwedToCountriesFromEu =
+      for {
+        allSales <- userAnswers.get(AllSalesWithTotalAndVatQuery).toList.flatten.zipWithIndex
```
Comment (hmrcvmu) [inline app/services/SalesAtVatRateService.scala:60]: Could use pattern matching and use:
```
(allSales, index) <- userAnswers.get(AllSalesWithTotalAndVatQuery).toList.flatten.zipWithIndex
salesFromCountry = allSales.country
salesToCountry <- userAnswers.get(AllSalesByCountryQuery(Index(index))).toSeq
```

**Review by hmrcvmu — DISMISSED** [review]

File: test/base/SpecBase.scala [hunk 4]
```diff
@@ -53,13 +55,30 @@ trait SpecBase
...
+    .set(VatRatesFromCountryPage(index, index), vatRates).success.value
+    .set(SalesToCountryPage(index, index), BigDecimal(100)).success.value
+    .set(VatOnSalesPage(index, index), VatOnSales(VatOnSalesChoice.Standard, BigDecimal(20))).success.value
+
+//  def completeUserAnswers: UserAnswers = completeSalesFromNIUserAnswers
+//    .set(SoldGoodsFromEuPage, true).success.value
+//    .set(CountryOfSaleFromEuPage(index), Country("HR", "Croatia")).success.value
+//    .set(VatRatesFromEuPage(index, index), List(twentyPercentVatRate)).success.value
+//    .set(NetValueOfSalesFromEuPage(index, index, index), BigDecimal(100)).success.value
+//    .set(VatOnSalesFromEuPage(index, index, index), VatOnSales(Standard, BigDecimal(20))).success.value
```
Comment (hmrcvmu) [inline test/base/SpecBase.scala:80]: We can remove this comment

File: test/controllers/CheckYourAnswersControllerSpec.scala [hunk 5]
```diff
@@ -17,46 +17,948 @@
...
+
+          status(result) mustEqual OK
+          contentAsString(result).contains("Business name") mustBe true
+          contentAsString(result).contains("UK VAT registration number") mustBe true
+          contentAsString(result).contains("Return month") mustBe true
+          contentAsString(result).contains("Sales to EU countries and Northern Ireland") mustBe true
+          contentAsString(result).contains("Sales made") mustBe true
+          contentAsString(result).contains("Sales excluding VAT") mustBe true
+          contentAsString(result).contains("Corrections") mustBe true
+//          contentAsString(result).contains("Corrections made") mustBe true
```
Comment (hmrcvmu) [inline test/controllers/CheckYourAnswersControllerSpec.scala:59]: We can remove this comment

File: test/controllers/CheckYourAnswersControllerSpec.scala [hunk 6]
```diff
@@ -17,46 +17,948 @@
...
```
Comment (hmrcvmu) [inline test/controllers/CheckYourAnswersControllerSpec.scala:83]: We can remove this comment

**Review by DevAndyroo — DISMISSED** [review]

**Review by DevAndyroo — APPROVED** [review]

**Pull request merged**